src/components/VirtualizedGrid/VirtualizedGrid.jsx

```diff
@@ -0,0 +1,341 @@
+      },
+      [getItemId]
+    );
+
+    // Memos
+    // Creates object of itemId => { item, index, height, offsetTop}
+    const normalizedItems = useMemo(() => {
+      return getNormalizedItems(items, idGetter, heightGetter);
+    }, [items, idGetter, heightGetter]);
+
+    const maxGridOffset = useMemo(() => {
+      return getMaxOffset(gridHeight, normalizedItems);
+    }, [gridHeight, normalizedItems]);
+
+    // Callbacks
+    const onScrollCB = useCallback(
+      ({ scrollDirection, scrollOffset, scrollUpdateWasRequested }) => {
+        scrollTopRef.current = scrollOffset;
+        if (!scrollUpdateWasRequested) {
+          animationData.scrollOffsetInitial = scrollOffset;
+        }
+        onScroll && onScroll(scrollDirection, scrollOffset, scrollUpdateWasRequested);
+      },
+      [onScroll, scrollTopRef, animationData]
+    );
+
+    const animateScroll = useCallback(() => {
+      requestAnimationFrame(() => {
+        const now = performance.now();
+        const ellapsed = now - animationData.animationStartTime;
+        const scrollDelta = animationData.scrollOffsetFinal - animationData.scrollOffsetInitial;
+        const easedTime = easeInOutQuint(Math.min(1, ellapsed / scrollDuration));
+        const scrollOffset = animationData.scrollOffsetInitial + scrollDelta * easedTime;
+        const finalOffsetValue = Math.min(maxGridOffset, scrollOffset);
+        scrollTopRef.current = finalOffsetValue;
+        gridRef.current.scrollTo(finalOffsetValue);
+
+        if (ellapsed < scrollDuration) {
+          animateScroll();
+        } else {
+          animationData.animationStartTime = undefined;
+          onScrollToFinished && onScrollToFinished();
+        }
+      });
+    }, [scrollDuration, animationData, gridRef, maxGridOffset, onScrollToFinished]);
+
+    const startScrollAnimation = useCallback(
+      item => {
+        const { offsetTop } = item;
+        if (animationData.animationStartTime) {
+          // animation already in progress
+          animationData.scrollOffsetFinal = offsetTop;
+          return;
+        }
+        if (animationData.scrollOffsetInitial === offsetTop) {
+          // offset already equals to item offset
+          onScrollToFinished && onScrollToFinished();
+          return;
+        }
+
+        animationData.scrollOffsetFinal = offsetTop;
+        animationData.animationStartTime = performance.now();
+        animateScroll();
+      },
+      [animationData, animateScroll, onScrollToFinished]
+    );
+
+    const cellRenderer = useCallback(
+      ({ columnIndex, rowIndex, style }) => {
+        const index = rowIndex * calcColumnCount + columnIndex;
+        const item = items[index];
+        return itemRenderer(item, index, style);
+      },
+      [items, itemRenderer, gridWidth, calcColumnCount]
+    );
+
+    const updateGridSize = useCallback(
+      (width, height) => {
+        if (height !== gridHeight || width !== gridWidth) {
+          setTimeout(() => {
```

Comment:
@MosheZemah why do we need a setTimeout 0 ?

Reply:
To avoid setting state while in render